From 90aee1d2d233b1df051cf41f77d5a092e335cb28 Mon Sep 17 00:00:00 2001
From: Philippe Normand <philn@igalia.com>
Date: Sun, 11 Apr 2021 13:27:56 +0100
Subject: [PATCH] i915: Prevent invalid framebuffer usage

When a surfaceless context is in use, driDrawablePriv might be NULL, so needs to
be checked before calling dri2InvalidateDrawable. Same for read calls.

Fixes #778
---
 src/mesa/drivers/dri/i915/intel_buffers.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/mesa/drivers/dri/i915/intel_buffers.c b/src/mesa/drivers/dri/i915/intel_buffers.c
index 83d59edb6fa..a4340fff6bb 100644
--- a/src/mesa/drivers/dri/i915/intel_buffers.c
+++ b/src/mesa/drivers/dri/i915/intel_buffers.c
@@ -62,7 +62,8 @@ intelDrawBuffer(struct gl_context * ctx)
        * time, invalidate our DRI drawable so we'll ask for new buffers
        * (including the fake front) before we start rendering again.
        */
-      dri2InvalidateDrawable(intel->driContext->driDrawablePriv);
+      if (intel->driContext->driDrawablePriv)
+         dri2InvalidateDrawable(intel->driContext->driDrawablePriv);
    }
 
    intel_draw_buffer(ctx);
@@ -79,7 +80,8 @@ intelReadBuffer(struct gl_context * ctx, GLenum mode)
        * time, invalidate our DRI drawable so we'll ask for new buffers
        * (including the fake front) before we start reading again.
        */
-      dri2InvalidateDrawable(intel->driContext->driReadablePriv);
+      if (intel->driContext->driReadablePriv)
+         dri2InvalidateDrawable(intel->driContext->driReadablePriv);
    }
 }
 
-- 
2.31.1

